sdk/src/Core/RuntimeDependencies/CreateInstanceContext.cs

```diff
@@ -0,0 +1,276 @@
+
+        /// <summary>
+        /// Data for the SSOClient context
+        /// </summary>
+        public SSOClientContext SSOClientContextData { get; }
+
+        /// <summary>
+        /// Data for the SOOIDCClient context
+        /// </summary>
+        public SSOOIDCClientContext SSOOIDCClientContextData { get; }
+    }
+
+
+    /// <summary>
+    /// Context for the factory method to create the SecurityTokenServiceClient runtime dependency.
+    /// </summary>
+    public class SecurityTokenServiceClientContext
+    {
+        /// <summary>
+        /// The possibly actions the the SecurityTokenServiceClient will be created for.
+        /// </summary>
+        public enum ActionContext { AssumeRoleAWSCredentials, AssumeRoleWithWebIdentityCredentials, FederatedAWSCredentials };
+
+        /// <summary>
+        /// Create an instance of the context
+        /// </summary>
+        /// <param name="action">The action the SecurityTokenServiceClient is being created for.</param>
+        /// <param name="region">The region the SDK expects the SecurityTokenServiceClient to be configured for.</param>
+        public SecurityTokenServiceClientContext(ActionContext action, RegionEndpoint region)
+        {
+            Action = action;
+            Region = region;
+        }
+
+        /// <summary>
+        /// The action the SecurityTokenServiceClient is being created for. For example AssumeRoleAWSCredentials.
+        /// </summary>
+        public ActionContext Action { get; }
+
+        /// <summary>
+        /// The region the SDK expects the SecurityTokenServiceClient to be configured for.
+        /// </summary>
+        public RegionEndpoint Region { get; }
+    }
+
+    /// <summary>
+    /// Context for the factory method to create the KeyManagementServiceClient runtime dependency.
+    /// </summary>
+    public class KeyManagementServiceClientContext
+    {
+        /// <summary>
+        /// Create an instance of the context
+        /// </summary>
+        /// <param name="parentServiceClient">The service client that is creating the KeyManagementServiceClient.</param>
+        public KeyManagementServiceClientContext(AmazonServiceClient parentServiceClient)
+        {
+            ParentServiceClient = parentServiceClient;
+        }
+
+        /// <summary>
+        /// The service client that is creating the KeyManagementServiceClient. The factory method should use the 
+        /// configuration like AWS region when constructing the KeyManagementServiceClient.
+        /// </summary>
+        public AmazonServiceClient ParentServiceClient { get; }
+    }
+
+    /// <summary>
+    /// Context for the factory method to create the SigV4aCrtSigner runtime dependency.
+    /// </summary>
+    public class SigV4aCrtSignerContext
+    {
+        /// <summary>
+        /// Create an instance of the context
+        /// </summary>
+        /// <param name="payload"></param>
+        public SigV4aCrtSignerContext(bool payload)
```

Comment:
should be `signPayload` not `payload`. The `param` in the comment needs filling out and corrected.

Reply:
Done